x-pack/plugins/apm/public/components/app/RumDashboard/URLFilter/URLSearch/index.tsx

```diff
@@ -0,0 +1,201 @@
+          params: {
+            query: {
+              start,
+              end,
+              uiFilters: JSON.stringify(restFilters),
+              urlQuery: searchValue,
+            },
+          },
+        });
+      }
+      return Promise.resolve(null);
+    },
+    [start, end, serviceName, uiFilters, searchValue]
+  );
+
+  useEffect(() => {
+    setCheckedUrls(uiFilters.transactionUrl || []);
+  }, [uiFilters]);
+
+  const onChange = (updatedOptions: EuiSelectableTemplateSitewideOption[]) => {
+    const clickedItems = updatedOptions.filter(
+      (option) => option.checked === 'on'
+    );
+
+    setCheckedUrls(clickedItems.map((item) => item.url));
+  };
+
+  // @ts-expect-error
+  const items: EuiSelectableTemplateSitewideOption[] = (data?.items ?? []).map(
+    (item) => ({
+      label: item.url,
+      key: item.url,
+      meta: [
+        {
+          text: I18LABELS.pageViews + ': ' + item.count,
+          type: 'deployment',
+          highlightSearchString: false,
+        },
+        {
+          text: I18LABELS.pageLoadDuration + ': ' + formatToSec(item.pld),
+          type: 'deployment',
+          highlightSearchString: false,
+        },
+      ],
+      url: item.url,
+      checked: checkedUrls?.includes(item.url) ? 'on' : null,
+    })
+  );
+
+  const onInputChange = (e: FormEvent<HTMLInputElement>) => {
+    setDebouncedValue(e.currentTarget.value);
+  };
+
+  const isLoading = status !== 'success';
+
+  const titleText = searchValue
+    ? (data?.total ?? 0) + ' ' + I18LABELS.searchResults
+    : I18LABELS.topPages;
+
+  function PopOverTitle() {
+    return (
+      <EuiFlexGroup>
+        <EuiFlexItem>
+          {isLoading ? <EuiLoadingSpinner /> : titleText}
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiButtonEmpty
+            size="s"
+            disabled={searchValue === ''}
+            onClick={() => {
+              updateSearchTerm(searchValue);
+              setForceClosePopover(true);
+            }}
+          >
+            {I18LABELS.matchThisQuery}
+          </EuiButtonEmpty>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    );
+  }
+
+  return (
+    <>
+      <EuiTitle size="xxs" textTransform="uppercase">
+        <h4>{I18LABELS.url}</h4>
+      </EuiTitle>
+      <EuiSelectableTemplateSitewide
+        data-test-subj="csmSearchFieldSuggestion"
+        singleSelection={false}
+        isLoading={isLoading}
+        onChange={onChange}
+        options={items}
+        searchProps={{
+          placeholder: I18LABELS.searchByUrl,
+          onInput: onInputChange,
+          inputRef: setSearchRef,
+        }}
+        popoverProps={{
+          closePopover: () => {
+            onFilterChange(checkedUrls);
+          },
+          ...(forceClosePopover ? { isOpen: false } : {}),
+        }}
+        popoverTitle={<PopOverTitle />}
+        listProps={{
+          showIcons: true,
+          onFocusBadge: {
```

Comment:
I am wondering if this badge is necessary. It seems obvious to me that the items in the list are clickable, and when they're clicked a checked icon appears and disappears to indicate selection. 🤷